### PR TITLE
Add generate random support into TPM provider

### PR DESCRIPTION
**Pull request by anta5010**
Implements #593

Signed-off-by: Anton Antonov <Anton.Antonov@arm.com>

### CODE REVIEW

File: src/providers/tpm/generate_random.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        let mut esapi_context = self
+            .esapi_context
+            .lock()
+            .expect("ESAPI Context lock poisoned");
+
+        let random_bytes = esapi_context
+            .as_mut()
+            .execute_without_session(|esapi_context| esapi_context.get_random(size))
+            .expect("Failed to get random bytes")
```
Comment (ionut-arm) [inline src/providers/tpm/generate_random.rs:22]: Probably better to convert this error and send it back up instead of panicking here.

**Review by ionut-arm — COMMENTED** [review]
Looks good, only got one request

File: src/providers/tpm/generate_random.rs (hunk 1 above)
Comment (anta5010) [reply src/providers/tpm/generate_random.rs]: updated

**Review by Kakemone — APPROVED** [review]

**Comment by ionut-arm** [comment]
cc @GowthamSureshKumar @MohamedAsaker

**Pull request merged**